launcher/ui/dialogs/BlockedModsDialog.cpp

```diff
@@ -1,28 +1,193 @@
+#include "Application.h"
 #include "BlockedModsDialog.h"
 #include "ui_BlockedModsDialog.h"
 #include <QPushButton>
 #include <QDialogButtonBox>
 #include <QDesktopServices>
 
+#include <QDebug>
+#include <QStandardPaths>
 
-BlockedModsDialog::BlockedModsDialog(QWidget *parent, const QString &title, const QString &text, const QString &body, const QList<QUrl> &urls) :
-        QDialog(parent), ui(new Ui::BlockedModsDialog), urls(urls) {
+
+
+
+BlockedModsDialog::BlockedModsDialog(QWidget *parent, const QString &title, const QString &text, QList<BlockedMod> &mods) :
+        QDialog(parent), ui(new Ui::BlockedModsDialog), mods(mods) {
     ui->setupUi(this);
 
     auto openAllButton = ui->buttonBox->addButton(tr("Open All"), QDialogButtonBox::ActionRole);
     connect(openAllButton, &QPushButton::clicked, this, &BlockedModsDialog::openAll);
 
+    connect(&watcher, &QFileSystemWatcher::directoryChanged, this, &BlockedModsDialog::directoryChanged);
+
+    hashing_task = shared_qobject_ptr<ConcurrentTask>(new ConcurrentTask(this, "MakeHashesTask", 10));
+    
+    qDebug() << "Mods List: " << mods;
+
+    setupWatch();
+    scanPaths();
+
     this->setWindowTitle(title);
     ui->label->setText(text);
-    ui->textBrowser->setText(body);
+    ui->labelModsFound->setText("Please download the missing mods.");
+    update();
 }
 
 BlockedModsDialog::~BlockedModsDialog() {
     delete ui;
 }
 
 void BlockedModsDialog::openAll() {
-    for(auto &url : urls) {
-        QDesktopServices::openUrl(url);
+    for(auto &mod : mods) {
+        QDesktopServices::openUrl(mod.websiteUrl);
+    }
+}
+
+/// @brief update UI with current status of the blocked mod detection
+void BlockedModsDialog::update() {
```

Comment:
add `tr()` to the user-facing english text in this function pls

Reply:
Is there a place I need to add the new strings for them to get added to Weblate when this is merged?

Reply:
nop, just adding `tr()` is enough

Reply:
Ok then, we should be all good on that then.